core/ArchiveProcessor/Loader.php

```diff
@@ -373,6 +374,44 @@ public function canSkipThisArchive()
         $isArchivingForcedWhenNoVisits = $this->shouldArchiveForSiteEvenWhenNoVisits();
         $hasSiteVisitsBetweenTimeframe = $this->hasSiteVisitsBetweenTimeframe($idSite, $params->getPeriod());
         $hasChildArchivesInPeriod = $this->dataAccessModel->hasChildArchivesInPeriod($idSite, $params->getPeriod());
+        $hasSegment = !$params->getSegment()->isEmpty();
+
+        if ($hasSegment) {
+            $periodEnd = $params->getPeriod()->getDateEnd();
+            $segmentHash = $params->getSegment()->getHash();
+            $segmentArchiving = StaticContainer::get(SegmentArchiving::class);
+            $segmentInfo = $segmentArchiving->findSegmentForHash($segmentHash, $idSite);
+
+            if ($segmentInfo) {
+                $segmentArchiveStartDate = $segmentArchiving->getReArchiveSegmentStartDate($segmentInfo);
+
+                if ($segmentArchiveStartDate->isLater($periodEnd)) {
+                    $doneFlag = Rules::getDoneStringFlagFor(
```

Comment:
not sure if we'd maybe need to use `getDoneFlags` or so and iterate over each done flag looking for the entries.

Reply:
we include the plugin creating this flag, as I tested it looked ok to me using this one and made more sense. but can't tell 100% sure